### PR TITLE
Fix MySQL `TINY` and `DATE` types support

### PR DESCRIPTION
**Pull request by ewgenius**
- **Handle MYSQL_TYPE_DATE**
- **Add Int8 mapping for arrow types**

Fixed missing MySQL `DATE` type, and added int8 (MySQL `TINYINT`) to arrow types mapping

Before fix (mysql table with tinyint and date columns):

CleanShot 2024-04-10 at 17 25 07@2x

CleanShot 2024-04-10 at 17 26 28@2x

After:

CleanShot 2024-04-10 at 17 27 10@2x

### CODE REVIEW

File: crates/arrow_sql_gen/src/mysql.rs [hunk 1]
```diff
@@ -265,6 +265,43 @@ pub fn rows_to_arrow(rows: &[Row]) -> Result<RecordBatch> {
...
+                                Value::Date(year, month, day, _, _, _, _) => {
+                                    chrono::NaiveDate::from_ymd_opt(
+                                        i32::from(year),
+                                        u32::from(month),
+                                        u32::from(day),
+                                    )
+                                    .unwrap_or_default()
+                                }
+                                _ => {
+                                    chrono::NaiveDate::from_ymd_opt(1970, 1, 1).unwrap_or_default()
```
Comment (phillipleblanc) [inline crates/arrow_sql_gen/src/mysql.rs]: When would this happen? It feels weird to default to 1970 for a date

File: crates/arrow_sql_gen/src/mysql.rs [hunk 2]
```diff
@@ -265,6 +265,43 @@ pub fn rows_to_arrow(rows: &[Row]) -> Result<RecordBatch> {
...
+                    match v.clone() {
+                        Some(v) => {
+                            let date = match v {
+                                Value::Date(year, month, day, _, _, _, _) => {
+                                    chrono::NaiveDate::from_ymd_opt(
+                                        i32::from(year),
+                                        u32::from(month),
+                                        u32::from(day),
+                                    )
+                                    .unwrap_or_default()
```
Comment (phillipleblanc) [inline crates/arrow_sql_gen/src/mysql.rs]: Why is the default value acceptable here?

**Comment by gloomweaver** [comment]
With chrono feature enabled for mysql_async can just do 
```
ColumnType::MYSQL_TYPE_DATE => {
                    let Some(builder) = builder else {
                        return NoBuilderForIndexSnafu { index: i }.fail();
                    };
                    let Some(builder) = builder.as_any_mut().downcast_mut::<Date32Builder>() else {
                        return FailedToDowncastBuilderSnafu {
                            mysql_type: format!("{mysql_type:?}"),
                        }
                        .fail();
                    };
                    let v = row.get_opt::<NaiveDate, usize>(i).transpose().context(
                        FailedToGetRowValueSnafu {
                            mysql_type: ColumnType::MYSQL_TYPE_DATE,
                        },
                    )?;

                    match v {
                        Some(v) => {
                            builder.append_value(Date32Type::from_naive_date(v));
                        }
                        None => builder.append_null(),
                    }
                }
```

File: crates/arrow_sql_gen/src/mysql.rs (hunk 1 above)
Comment (ewgenius) [reply crates/arrow_sql_gen/src/mysql.rs]: oh, you're right. fixed to set `null` instead

File: crates/arrow_sql_gen/src/mysql.rs (hunk 2 above)
Comment (ewgenius) [reply crates/arrow_sql_gen/src/mysql.rs]: refactored

**Review by phillipleblanc — DISMISSED** [review]

File: test/scripts/setup-data-mysql.sql [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+--   NULL,
+--   NULL,
+--   NULL,
+--   NULL,
+--   NULL,
+--   NULL,
+--   NULL,
+--   NULL,
+--   NULL
+-- );
```
Comment (ewgenius) [inline test/scripts/setup-data-mysql.sql:63]: tracked as a bug in #1066

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**